wordpress-mobile/WordPress-Android
# [JP Install Full Plugin] Implement tracking on install flow

Fixes #17835 

To test:
1 - Install app and select site with individual plugin(s);
2 - Open home screen: the JP full plugin install card should be shown with "Learn more" in the card and "Hide this" in the overflow menu;
3 - Tap on "Learn more" to open the onboarding screen;
4 - Tap on "Install the full plugin" button to open the install flow;

5 - When install flow is opened with initial state: track `"jp_install_full_plugin_flow_viewed"` with parameter `{ "status": "initial" }`;
6 - When Toolbar back button is tapped with initial state: track `"jp_install_full_plugin_flow_cancel_tapped"` with parameter `{ "status": "initial" }`;
7 - When the system navigation back button is tapped with initial state: track `"jp_install_full_plugin_flow_cancel_tapped"` with parameter `{ "status": "initial" }`;
8 - When loading state is shown: track `"jp_install_full_plugin_flow_viewed"` with parameter `{ "status": "loading" }`;
9 - When "Continue" button is clicked: track `"jp_install_full_plugin_flow_install_tapped"`
10 - When Toolbar back button is tapped with loading state: track `"jp_install_full_plugin_flow_cancel_tapped"` with parameter `{ "status": "loading" }`;
11 - When the system navigation back button is tapped with loading state: track `"jp_install_full_plugin_flow_cancel_tapped"` with parameter `{ "status": "loading" }`;
12 - When JP installation succeeds: track `"jp_install_full_plugin_flow_success"`;
13 - When error state is shown: track `"jp_install_full_plugin_flow_viewed"` with parameter `{ "status": "error }`;
14 - When Toolbar back button is tapped with error state: track `"jp_install_full_plugin_flow_cancel_tapped"` with parameter `{ "status": "error" }`;
15 - When the system navigation back button is tapped with error state: track `"jp_install_full_plugin_flow_cancel_tapped"` with parameter `{ "status": "error" }`;

## Regression Notes
1. Potential unintended areas of impact
None.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual testing and unit testing.

3. What automated tests I added (or what prevented me from doing so)
Add `JetpackFullPluginInstallAnalyticsTrackerTest` and updated `JetpackFullPluginInstallViewModelTest`.

PR submission checklist:
